### DISCUSSION TITLE
Question regarding duplicate resources

### DISCUSSION THREAD

**Discussion by 0xd3e** [opentelekomcloud/terraform-provider-opentelekomcloud#1188] [Q&A] [answered]
Hi,

I wonder why there are some resources that are doing basically the same, but have a different name and a slightly different spec.

Examples are:
- opentelekomcloud_images_image_v2
- opentelekomcloud_ims_image_v2
- opentelekomcloud_compute_instance_v2
- opentelekomcloud_ecs_instance_v1

Is there some recommendation which one I should use to create a certain resource?

**Comment by outcatcher** [comment, marked as answer]
For `opentelekomcloud_compute_instance_v2` and `opentelekomcloud_ecs_instance_v1`:
- `opentelekomcloud_compute_instance_v2` uses OpenStack-compatible API
- `opentelekomcloud_ecs_instance_v1` uses OTC-native API
Generally, the choice between them depends on how are you accustomed to OpenStack. On the pros of the `ecs_instance_v1` - sometimes when server creation fails on some late stage, using `compute_instance_v2` will leave faulty instance behind, and `ecs_instance_v1` will remove it.

For images difference again in the API used:
- `ims_image` (OTC-specific): https://docs.otc.t-systems.com/en-us/api/ims/en-us_topic_0135460761.html
- `image_image` (Native OpenStack): https://docs.otc.t-systems.com/en-us/api/ims/en-us_topic_0122410334.html
In most cases, OTC-specific APIs contain more information. But I myself prefer OpenStack resources :slightly_smiling_face:

**Reply by 0xd3e** [reply]
Thanks for the explanation. I thought of that, but wasn't sure.

I get the idea behind it, but it is a bit confusing when some resources are dealing with lets say a VPC ID, which is a term you can find easily in the OTC console and the other one is using a network ID, which is IIRC not the same as the VPC ID.

So someone might have a hard time understanding all that and of course get all the necessary information because the OTC console is hiding a lot of information.

**Reply by outcatcher** [reply]
That's the most common confusion :slightly_smiling_face: 
OTC | OpenStack 
--- | --- 
VPC | Router
Subnet | Network + Subnet

We're planning to get rid of such confusion in the provider `v2.0`.
